mypyc/test-data/run-bytes.test

```diff
@@ -52,4 +52,21 @@ def test_len() -> None:
     # Use bytes() to avoid constant folding
     b = b'foo' + bytes()
     assert len(b) == 3
-    assert len(bytes()) == 0
\ No newline at end of file
+    assert len(bytes()) == 0
+
+[case testBytearrayBasics]
+
+def test_init() -> None:
+    brr1 = bytearray(3)
+    assert brr1 == bytearray(b'\x00\x00\x00')
+    assert brr1 == b'\x00\x00\x00'
+    l = [10, 20, 30, 40]
+    brr2 = bytearray(l)
+    assert brr2 == bytearray(b'\n\x14\x1e(')
+    assert brr2 == b'\n\x14\x1e('
+    brr3 = bytearray(range(5))
+    assert brr3 == bytearray(b'\x00\x01\x02\x03\x04')
+    assert brr3 == b'\x00\x01\x02\x03\x04'
+    brr4 = bytearray('string', 'utf-8')
+    assert brr4 == bytearray(b'string')
+    assert brr4 == b'string'
```

Comment:
Explicitly annotate the variables as `bytes`, since right now the inferred type will be `bytearray`, which doesn't have a mypyc primitive type, so it reverts back to `object`.

Also test calling a function that accepts `bytes` with an actual argument that is a `bytearray` object annotated with type `Any`, to check the runtime type checking from `Any` to `bytes` works if the runtime value is a `bytearray` .